src/Detector/Go/GoDetector.cs

```diff
@@ -0,0 +1,103 @@
+        private readonly DetectorOptions _options;
+
+        /// <summary>
+        /// Creates an instance of <see cref="GoDetector"/>.
+        /// </summary>
+        /// <param name="logger">The <see cref="ILogger{GoDetector}"/>.</param>
+        /// <param name="options">The <see cref="DetectorOptions"/>.</param>
+        public GoDetector(ILogger<GoDetector> logger, IOptions<DetectorOptions> options)
+        {
+            _logger = logger;
+            _options = options.Value;
+        }
+
+        public PlatformDetectorResult Detect(DetectorContext context)
+        {
+            bool isGoApp = false;
+            string appDirectory = string.Empty;
+            var sourceRepo = context.SourceRepo;
+            bool goDotModExists = false;
+            // check if go.mod exists
+            if (sourceRepo.FileExists(GoConstants.GoDotModFileName))
+            {
+                goDotModExists = true;
+                isGoApp = true;
+                _logger.LogInformation($"Found {GoConstants.GoDotModFileName} at the root of the repo. ");
+            }
+            else
+            {
+                _logger.LogDebug(
+                    $"Could not find {GoConstants.GoDotModFileName} in repo");
+                return null;
+            }
+
+            // TODO: check if go.sum or /cmd/repo_name/*.go exists
+
+            var version = GetVersion(context);
+            return new GoPlatformDetectorResult
+            {
+                Platform = GoConstants.PlatformName,
+                PlatformVersion = version,
+                AppDirectory = appDirectory,
+            };
+        }
+
+        private string GetVersion(DetectorContext context)
+        {
+            var versionFromGoDotMod = GetVersionFromGoDotMod(context);
+            if (versionFromGoDotMod != null)
+            {
+                return versionFromGoDotMod;
+            }
+            return null;
+        }
+
+        private string GetVersionFromGoDotMod(DetectorContext context)
+        {
+            try
+            {
+                var goDotModFileContent = context.SourceRepo.ReadFile(GoConstants.GoDotModFileName);
+                var goDotModFileContentLines = goDotModFileContent.Split('\n');
+
+                // Example content of go.mod:
+                // module myModule
+                //
+                // go 1.16
+                foreach (var goDotModFileContentLine in goDotModFileContentLines)
+                {
+                    if (goDotModFileContentLine.Trim().StartsWith("go", StringComparison.InvariantCultureIgnoreCase))
+                    {
+                        var goVersionLine = goDotModFileContentLine.Trim().Split(' ');
+                        // Make sure it's in valid format
+                        if (goVersionLine.Length == 2)
+                        {
+                            return goVersionLine[1].Trim('\"').Trim('\'');
+                        }
+                    }
```

Comment:
I'm in the boat that it would be better to have the RegEx check here to be a little more safer about the `go MAJOR.MINOR` pattern.

Reply:
Good point, I'll look into that